multiversion/src/main/scala/multiversion/commands/ResolveCommand.scala

```diff
@@ -0,0 +1,57 @@
+package multiversion.commands
+
+import java.nio.file.Path
+import java.nio.file.Paths
+
+import moped.annotations.CommandName
+import moped.annotations.Description
+import moped.annotations.ExampleUsage
+import moped.annotations.PositionalArguments
+import moped.annotations.Repeated
+import moped.cli.Application
+import moped.cli.CommandParser
+import moped.json.Result
+import moped.reporters.Diagnostic
+
+@ExampleUsage("multiversion resolve //aaa/bbb:ccc")
+@CommandName("resolve")
+@Description("Query the classpath resolution given a target")
+case class ResolveCommand(
+    @Description("Print dependency tree") override val tree: Boolean = false,
+    @Description("Print //3rdparty references") `3rdparty`: Boolean = false,
+    @Description("Path to bazel executable") bazel: Path = Paths.get("bazel"),
+    @Repeated @Description("Additional arguments to pass to bazel") bazelArgs: List[String] = Nil,
+    @PositionalArguments targets: List[String] = Nil,
+    app: Application = Application.default
+) extends QueryBasedCommand {
+
+  override def runResult(): Result[Unit] =
+    for {
+      r <- runCustomQuery()
+    } yield (if (tree) r else r.sorted).foreach(Console.out.println(_))
+
+  private def resolveQuery(target: String): String =
+    s"""filter("@maven.*", kind("generated file", deps(${target})))"""
+
+  private def resolve3rdpartyQuery(target: String): String =
+    s"""kind("third_party_jvm_import", deps(${target}))"""
```

Comment:
Since this cannot be tested in this repo, what do you think of using a filter such as `filter("//3rdparty/jvm/.*", deps(${target}))`? Would that not work?

Reply:
Looks like your query would catch aliases too, which I think is a plus.